django_celery_results/backends/database.py

```diff
@@ -246,7 +246,7 @@ def on_chord_part_return(self, request, state, result, **kwargs):
         if not gid or not tid:
             return
         call_callback = False
-        with transaction.atomic():
+        with transaction.atomic(using=ChordCounter.objects.db):
```

Comment:
`BaseManager.objects.db` returns the `django.db.router.db_for_read()` which in your situation works since it returns the same db alias as the `db_for_write()`.

https://github.com/django/django/blob/2be37b253341cfd1f1363c533e6f896230f047a7/django/db/models/manager.py#L143

Consider using `django.db.router.db_for_write(ChordCounter)` instead as we will be updating the `ChordCounter.count` value within the transaction.